ingest/indexer/service/indexer_streaming_service.go

```diff
@@ -130,15 +156,16 @@ func (s *indexerStreamingService) publishTxn(ctx context.Context, req abci.Reque
 
 		// Publish the transaction
 		txn := domain.Transaction{
-			Height:             uint64(sdkCtx.BlockHeight()),
-			BlockTime:          sdkCtx.BlockTime().UTC(),
-			GasWanted:          uint64(gasWanted),
-			GasUsed:            uint64(gasUsed),
-			Fees:               fee,
-			MessageType:        msgType,
-			TransactionHash:    txHash,
-			TransactionIndexId: txnIndex,
-			Events:             includedEvents,
+			Height:              uint64(sdkCtx.BlockHeight()),
+			BlockTime:           sdkCtx.BlockTime().UTC(),
+			GasWanted:           uint64(gasWanted),
+			GasUsed:             uint64(gasUsed),
+			Fees:                fee,
+			MessageType:         msgType,
+			TransactionHash:     txHash,
+			TransactionIndexId:  txnIndex,
+			TokenInBeforeSpread: tokenInBeforeSpread,
```

Comment:
What if there are multiple swap messages in the same transaction?

Also, how do we handle swaps outside of transactions such as protorev?